src/test/storage/adaptive_radix_tree_index_test.cpp

```diff
@@ -72,8 +72,12 @@ class AdaptiveRadixTreeIndexTest : public BaseTest {
         EXPECT_EQ(index->upper_bound({*search_value}), index->lower_bound({*search_value}));
       }
 
-      int32_t min = **distinct_values.begin();
-      int32_t max = **distinct_values.rbegin();
+      const auto begin = distinct_values.begin();
+      const auto rbegin = distinct_values.rbegin();
+
+      Assert(*begin && *rbegin, "Optional has no value.");
```

Comment:
The reference says: "The behavior is undefined if *this does not contain a value." I think we have to use `has_value()` here.

Reply:
But in this line, we don't use `std::optional<T>::operator*`. We dereference the iterators `begin` and `rbegin` to get  the `optional<int32_t>` values they are pointing to.
Building the logical expression `*begin && *rbegin` converts the optional-values to bool-values.

cppreference:
>When an object of type optional<T> is contextually converted to bool, the conversion returns true if the object contains a value and false if it does not contain a value.

Pleas correct me if I'm wrong.